content/05-expression.md

```diff
@@ -798,14 +818,184 @@ If during runtime the evaluation of `try-expression` causes a [`throw`](expressi
+```
+Main.hx:13: 
+Called from Main.doSomething (Main.hx line 11 column 15)
+Called from Main.main (Main.hx line 5 column 5)
+Main.hx:17: 
+Called from Main.doSomething (Main.hx line 11 column 15)
+Called from Main.main (Main.hx line 5 column 5)
+```
+
+If `catch(e:haxe.Exception)` catches a native exception, then rethrowing `e` throws that native exception instead of a `haxe.Exception` instance.
+Here's an example of Haxe code, which being compiled to PHP target catches and rethrows all exceptions in the inner `try/catch`. And rethrown exceptions are still catchable using their target native types:
+```haxe
+try {
+  try {
+    (null:Dynamic).callNonExistentMethod();
+  } catch(e:Exception) {
+    trace('Haxe exception: ' + e.message);
+    throw e; //rethrow
+  }
+} catch(e:php.ErrorException) {
+  trace('Rethrown native exception: ' + e.getMessage());
+}
+```
+This sample being compiled to PHP target would print:
+```
+Main.hx:9: Haxe exception: Trying to get property 'callNonExistentMethod' of non-object
+Main.hx:13: Rethrown native exception: Trying to get property 'callNonExistentMethod' of non-object
+```
+
+#### Chaining exceptions
 
+##### Since Haxe 4.1
+
+Sometimes it's convenient to chain exceptions instead of throwing the same exception instance again.
+To do so just pass an exception to a new exception instance:
+```haxe
+try {
+  doSomething();
+} catch(e:haxe.Exception) {
+  cleanup();
+  throw new haxe.Exception('Failed to do something', e);
+}
+```
+Being executed with `--interp` this sample would print a message like this:
+```
+Main.hx:12: characters 7-12 : Uncaught exception Exception: Terrible error
+Called from Main.doSomething (Main.hx line 10 column 13)
+
+Next Exception: Failed to do something
+Called from Main.doSomething (Main.hx line 12 column 13)
+Called from Main.main (Main.hx line 5 column 5)
+Main.hx:5: characters 5-18 : Called from here
+```
+One use-case is to make error logs more readable.
+
+Chained exceptions are available through `previous` property of `haxe.Exception` instances:
+```haxe
+try {
+  try {
+    doSomething();
+  } catch(e:haxe.Exception) {
+    cleanup();
+    throw new haxe.Exception('Failed to do something', e);
+  }
+} catch(e:haxe.Exception) {
+  trace(e.message); // "Failed to do something"
+  trace(e.previous.message); // "Terrible error"
+}
+```
+
+Another use-case is creating a library, which does not expose internal exceptions to public API, but still provides information about exceptions reasons:
+```haxe
+import haxe.Exception;
+
+class MyLibException extends Exception {}
+
+class MyLib {
+  static public function calculateSomething() {
+    try {
+      heavyCalculation();
+    } catch(e:Exception) {
+      throw new MyLibException(e.message, e);
+    }
+  }
+
+  static function heavyCalculation() {}
+}
+```
+Now library users don't have to bother about specific arithmetic exceptions. All they need to do is handle `MyLibException`
```

Comment:
I'd phrase this as "don't have to _worry_ about" instead. Also, missing a dot at the end. :)